.clang-format

```diff
@@ -0,0 +1,26 @@
+﻿---
+AlignAfterOpenBracket: Align
+AlignConsecutiveAssignments: 'false'
+AlignConsecutiveDeclarations: 'false'
+AlignOperands: 'true'
+AlignTrailingComments: 'true'
+AllowAllParametersOfDeclarationOnNextLine: 'true'
+AllowShortBlocksOnASingleLine: 'false'
+AllowShortCaseLabelsOnASingleLine: 'false'
+AllowShortFunctionsOnASingleLine: Inline
+AllowShortIfStatementsOnASingleLine: 'false'
+AlwaysBreakAfterReturnType: All
+BinPackArguments: 'false'
+BreakBeforeBraces: Linux
+DerivePointerAlignment: 'false'
+IndentCaseLabels: 'false'
+IndentWidth: '8'
+Language: Cpp
+MaxEmptyLinesToKeep: '1'
+PointerAlignment: Right
+SortIncludes: 'true'
+SpaceBeforeParens: Always
```

Comment:
IIRC, this was the point that made @hughsie to reject it when I proposed it :)

Reply:
I do remember we had talked about this before, and specifically rejected a pre-commit hook for it.   Was there contention specifically about `SpaceBeforeParens` though?

I want to discuss it again though because I really think that it will make code review less painful and unify style across the whole project.  There's a few instances of minor things that are different in different plugins (such as return type on a separate line for functions for example).

I think we can easily have a CI task that runs clang-format and fail the job if there is a diff.  Then submitter just needs to run clang-format themselves and re-push.  Makes it really a lower barrier to entry to submitting.

So yes, it might require a few project wide style changes if we can't get a clang option for something specifically but I still think it's worth it.

Reply:
The discussion started [here](https://github.com/hughsie/fwupd/pull/201#discussion_r135654976), where @gicmo offered a baseline file to build on.

The rest was probably on an email thread between us three (@hughsie and both of us), but it was from my Intel address so I have no access for it anymore... Maybe one of you can still find it.
I remember this was one point that Richard didn't liked, and I linked there a TODO from clang-format code that there is a plan to add a mode that is more suitable to the current project style.